test/index.ts

```diff
@@ -0,0 +1,93 @@
+const hre = require("hardhat");
+const { expect } = require("chai");
+const { MerkleTree } = require("merkletreejs");
+const keccak256 = require("keccak256");
+
+const sampleData: { string: string } = require("./sample-data.json");
+
+async function deploy(name: any, ...params: any) {
+  const Contract = await hre.ethers.getContractFactory(name);
+  return await Contract.deploy(...params).then((f: any) => f.deployed());
+}
+
+function hashToken(account: string, value: string) {
+  return Buffer.from(
+    hre.ethers.utils
+      .solidityKeccak256(["address", "uint256"], [account, value])
+      .slice(2),
+    "hex"
+  );
+}
+
+describe("Create ERC20 Merkle Drop and test", function () {
+  let factory: any;
+  before(async function () {
+    factory = await deploy("MerkleFactory");
+    this.accounts = await hre.ethers.getSigners();
+    this.merkleTree = new MerkleTree(
+      Object.entries(sampleData).map((data: any) =>
+        hashToken(data[0], data[1])
+      ),
+      keccak256,
+      { sortPairs: true }
+    );
+  });
+
+  describe("Claim tokens", function () {
+    let deployedToken: any;
+    let contractOne: any;
+    before(async function () {
+      const erc20DropTemplate = await deploy("MerkleDrop");
+      // Test token
+      deployedToken = await deploy("MyToken");
+      // Create a new contract from factory contract
+      await factory.createDrop(
+        erc20DropTemplate.address,
+        deployedToken.address,
+        this.merkleTree.getHexRoot(),
+        Date.now() + 1000000
+      );
+      const createdMerkleDrops = await factory.getAllMerkleDrops();
+      contractOne = createdMerkleDrops[0];
+      console.log("🚀 ~ file: index.ts ~ line 53 ~ contractOne", contractOne);
```

Comment:
WDYM by `contractOne`?

Reply:
The cloned contract

Reply:
Can you please rename it to a meaningful name 🙇🏻  so other contributors can understand it!

Reply:
Updated better name